File: java/socketd-broker-test/src/test/java/features/cases/BaseTestCase.java
package features.cases;

/**
 * @author noear
 * @since 2.0
 */
public abstract class BaseTestCase {
    private final String schema;
    private final int port;

    public String getSchema() {
        return schema;
    }

    public int getPort() {
        return port;
    }

    public BaseTestCase(String schema, int port) {
        this.schema = schema;
        this.port = port;
    }

    /**
     * 开始测试
     * */
    public abstract void start() throws Exception;

    /**
     * 停止测试
     * */
    public abstract void stop() throws Exception;

    public void onError(){
        System.out.println("--------test error: " + getSchema());
    }
}

File: python/socketd/test/modelu/BaseTestCase.py
from loguru import logger

class BaseTestCase:

    def __init__(self, schema, port):
        self.schema = schema
        self.port = port

    def start(self):
        logger.info (f"--------START {self.schema}------")

    def stop(self):
        logger.info(f"--------STOP {self.schema}------")

    def on_error(self):
        logger.info(f"--------ERROR {self.schema}------")
